front-end/src/app/shared/models/transaction-types/SchaTransactionType.model.ts

```diff
@@ -0,0 +1,12 @@
+import { TransactionType } from '../../interfaces/transaction-type.interface';
+
+export abstract class SchaTransactionType extends TransactionType {
+  override generatePurposeDescriptionLabel?(): string {
+    if (this.generatePurposeDescription !== undefined) {
+      return '(SYSTEM-GENERATED)';
+    } else if (!this.purposeDescriptionUserInputRequired) {
```

Comment:
I thought the point of adding this class was to check the schema for whether or not the description is required
If we don't do that, we still have to re-configure `purposeDescriptionUserInputRequired` in every model
```
if (this.generatePurposeDescription) {
  return '(SYSTEM-GENERATED)';
} else if (transactionType?.schema.required.includes('contribution_purpose_descrip')) {
  return '';
}
return '(OPTIONAL)';

Reply:
@toddlees yes I forgot to reach into the schema while focused on the x2-base fix.  Went ahead and added that and pushed it.  Thanks for the snippet.